### PR TITLE
Add support to show a single direct message.

### PR DESCRIPTION
**Pull request by waseem**
I was not able to test this though. When I run tests, I get following error.

```
 1) test shows direct message (ExTwitterTest)
     test/extwitter_test.exs:101
     ** (ExVCR.RequestNotMatchError) Request did not match with any one in the current cassette: fixture/vcr_cassettes/show_direct_message.json.
     Delete the current cassette with [mix vcr.delete] and re-record.
     
     stacktrace:
       lib/exvcr/handler.ex:116: ExVCR.Handler.raise_error_if_cassette_already_exists/1
       lib/exvcr/handler.ex:100: ExVCR.Handler.get_response_from_server/2
       (inets) :httpc.request(:get, {'https://api.twitter.com/1.1/direct_messages/show/446328507694845952.json?oauth_signature=ZyUwlXCAMj8CXyr2V6kSfHcEOjo%3D&oauth_version=1.0&oauth_nonce=fVfwbwczFr6Lf6h7K9MppaoK%2FpdE7NvZqUUyYjQTM%2Bg%3D&oauth_timestamp=1446520933&oauth_signature_method=HMAC-SHA1&oauth_consumer_key=nil&oauth_token=nil', []}, [autoredirect: false], [])
       (extwitter) lib/extwitter/api/base.ex:26: ExTwitter.API.Base.do_request/3
       (extwitter) lib/extwitter/api/direct_messages.ex:10: ExTwitter.API.DirectMessages.direct_message/2
       test/extwitter_test.exs:103
```

I do not know id of a direct message that I could use in tests. How will I go with testing it?

cc: @parroty

### CODE REVIEW

**Comment by parroty** [comment]
Thanks, great! 

For the testing part, it seems empty `fixture/vcr_cassettes/show_direct_message.json` is registered.  `mix vcr.delete show_direct_message` and retry testing (`mix test`) might resolve the error. I couldn't access to the specific direct message (446328507694845952) due to permission, and I might want to tweak it anyway.

I just tested with `ExTwitter.request(:get, "1.1/direct_messages.json")` for identifying direct message id to test, but I'm wondering it's good to have `ExTwitter.direct_messages` too. Did you find any catch on this API? Otherwise, I'm thinking to add along with this PR.

File: lib/extwitter.ex [hunk 1]
```diff
@@ -543,6 +543,19 @@ defmodule ExTwitter do
...
+  GET direct_messages/show/:id
+
+  ## Examples
+
+      ExTwitter.direct_message(446328507694845952)
+
+  ## Reference
+  https://dev.twitter.com/rest/reference/get/direct_messages/show
+  """
+  @spec show(Integer | String.t) :: ExTwitter.Model.DirectMessage.t
```
Comment (parroty) [inline lib/extwitter.ex]: Maybe it's `direct_message` instead of `show`

File: fixture/vcr_cassettes/show_direct_message.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (parroty) [inline fixture/vcr_cassettes/show_direct_message.json]: Maybe `mix vcr.delete show_direct_message` will delete this file and re-executing the test records the information.

**Comment by waseem** [comment]
@parroty I was having trouble accessing acquiring consumer key, secret and access token and secret because I don't have access to any twitter application. I tried to create one though. But creating one requires me to add my phone and when I try to add phone to my twitter account, twitter does not send me a verification sms. 

Anyways, I used https://dev.twitter.com/rest/tools/console to get sample response for the API endpoint and wrote the test based on response received.

I do plan to add `ExTwitter.direct_messages` with this PR. I was getting a hang of the code and testing process with this initial implementation. Wait for a little more for other part.

**Comment by waseem** [comment]
@parroty Aright! I also added `ExTwitter.direct_messages`. Let me know once you're done with code review. I'll squash the commits so that you can merge.

**Comment by parroty** [comment]
@waseem Thanks. The updates seems good. I can merge, but shall I wait for the squash?

**Comment by waseem** [comment]
@parroty I squashed the commits.

**Comment by parroty** [comment]
Thanks! :+1:

**Comment by parroty** [comment]
Pushed as v0.5.3. 
Just a note, but I added ExTwitter.direct_messages/0 since it's defined in ExTwitter.API.DirectMessages, but was not exposed to ExTwitter module.

**Pull request merged**